File: backend/src/CreditFinancialOrganization.Domain/Entities/Loans/LoanApplication.cs
using CreditFinancialOrganization.Domain.Entities.Users;
using CreditFinancialOrganization.Domain.Enums;

namespace CreditFinancialOrganization.Domain.Entities.Loans;

public class LoanApplication : BaseEntity
{
    public DateTime Date { get; set; }
    public ApplicationStatus Status { get; set; }
    public DateOnly? ApprovalDate { get; set; }
    public Guid? EmployeeId { get; set; }

    public Loan Loan { get; set; } = null!;
    public User? Employee { get; set; }
}

File: frontend/src/app/shared/models/loans/loan-application.ts
import { ApplicationStatus } from "../../enums/application-status";
import { User } from "../users/user";
import { Loan } from "./loan";

export interface LoanApplication {
    id: string;
    date: Date;
    status: ApplicationStatus;
    approvalDate?: Date;
    loan: Loan;
    employee?: User;
}
